src/components/challenges/active.tsx

```diff
@@ -27,26 +30,49 @@ type ActiveProps = {
     }[];
     is_active: boolean;
     is_user_subscribed: number;
-  };
+  },
+  userId: string
 };
 
-const Active: FC<ActiveProps> = ({ content }) => {
+const Active: FC<ActiveProps> = ({ content, userId }) => {
   const [isUserSubscribed, setUserSubscribed] = useState(content.is_user_subscribed);
 
+  const subscibeUser = () => {
+    fetch(SUBSCRIBE_TO_CHALLENGE_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        challenge_id: content.id,
+        user_id: userId,
+      }),
+      credentials: 'include',
+    })
+      .then((res) => res.json())
+      .then((res) => setUserSubscribed(res.is_user_subscribed))
+      .catch((err) => {
+        toast.error(err, {
+          position: 'top-center',
+          autoClose: 4000,
+          hideProgressBar: false,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: false,
+          progress: undefined,
+        });
```

Comment:
This needs to be placed outside of this. what if we want to use the toast elsewhere? are we going to write this again?

Instead create a function that calls this, and invoke this function with the error message.

Reply:
Yeah, we need to get someone to build these toast functions.
I'll create an issue 👍🏽 
Thanks